pkg/controller/plan/adapter/ova/builder.go

```diff
@@ -510,27 +518,36 @@ func getDiskSourcePath(filePath string) string {
 }
 
 func getResourceCapacity(capacity int64, units string) (int64, error) {
-	if units == "" {
-		return 0, nil
-	}
-
-	re := regexp.MustCompile("[0-9]+")
-
-	numbers := re.FindAllString(units, -1)
-	if len(numbers) != 2 {
-		return 0, nil
-	}
-	base, err := strconv.Atoi(numbers[0])
-	if err != nil {
-		return 0, err
-	}
-	pow, err := strconv.Atoi(numbers[1])
-	if err != nil {
-		return 0, err
+	items := strings.Split(units, "*")
+	for i := range items {
+		item := strings.TrimSpace(items[i])
+		if i == 0 {
+			if item != "byte" {
+				return 0, errors.New(fmt.Sprintf("units '%s' are invalid, only 'byte' is supported", units))
+			}
+		} else {
+			num, err := strconv.Atoi(item)
+			if err == nil {
+				capacity = capacity * int64(num)
+				continue
+			}
+			nums := strings.Split(item, "^")
+			if len(nums) == 2 {
+				base, err := strconv.Atoi(nums[0])
+				if err != nil {
+					return 0, errors.New(fmt.Sprintf("units '%s' are invalid, base component is invalid: %s", units, item))
+				}
+				pow, err := strconv.Atoi(nums[1])
+				if err != nil {
+					return 0, errors.New(fmt.Sprintf("units '%s' are invalid, pow component is invalid: %s", units, item))
+				}
+				capacity = capacity * int64(math.Pow(float64(base), float64(pow)))
+				continue
+			}
+			return 0, errors.New(fmt.Sprintf("units '%s' are invalid, item is invalid: %s", units, item))
+		}
```

Comment:
@bkhizgiy I didn't verify these changes, please take a look

Reply:
@ahadas I see that the test failing, so I"ll take a look whats missing.

Reply:
right, I think I've fixed that test that failed - `byte * 2` is supposed to be a valid input